script.js

```diff
@@ -1,12 +1,543 @@
+  let select = document.getElementById("select");
+  if (select !== null) {
+    for (let index = select.options.length; index > 0; index--) {
+      select.remove(index);
+    }
+
+    episodeList.forEach((episode) => {
+      let option = document.createElement("option");
+      option.innerHTML = `${episode.name}: S${formatSeasonNum(
+        episode.season
+      )}E${formatSeasonNum(episode.number)}`;
+      select.appendChild(option);
+    });
+  }
+}
+
+const addDropDownMenu = (episodeList) => {
+  // select on the right  // checks if the drop down exist it creates it else it removes the options 222
+  let option;
+
+  // this condition prevents duplication
+  select = document.createElement("select");
+  select.style.display = "none"; // hide select episodes on the tv show listing
+
+  select.setAttribute("id", "select");
+  option = document.createElement("option");
+  option.innerText = "All episodes";
+  select.appendChild(option);
+
+  select.addEventListener("change", function (event) {
+    let episodeContainers = document.getElementsByClassName(
+      "episode-container-class-name"
+    ); // An array of episode container
+
+    let episodeCounter = 0;
+    let containers = Array.from(episodeContainers);
+
+    if (event.target.options[event.target.selectedIndex].index === 0) {
+      // refers to the all episode option
+      containers.forEach((container) => {
+        episodeCounter = containers.length;
+        container.style.display = "";
+      });
+    } else {
+      containers.forEach((container, episodeIndex) => {
+        if (
+          episodeIndex ===
+          event.target.options[event.target.selectedIndex].index - 1 // test if the current index is equal to the index
+        ) {
+          showEpisode(container); //.style.display = "";
+        } else {
+          hideEpisode(container); //.style.display = "none";
+        }
+        // This increments until it matches the episode index});
+      });
+      episodeCounter = 1; // It keeps counting until it reaches the end of the array
+    }
+    updateCounter(episodeCounter, containers.length);
+  });
+  return select;
+};
+
+// counter function display creates the dom
+const addCounter = (_) => {
+  counterDisplay = document.createElement("div");
+  counterDisplay.setAttribute("id", "counterDisplayID");
+  counterDisplay.style.marginTop = "0.5em"; // centre below
+
+  return counterDisplay;
+};
+
+function updateCounter(numberOfMovies, totalMovies) {
+  // counter for number of episode found in search
+  let tvSelectDom = document.querySelector("#select-tv");
+  let tvShowDisplay;
+  if (tvSelectDom.style.display === "none") {
+    tvShowDisplay = "Episodes";
+  } else {
+    tvShowDisplay = "TvShow";
+  }
+
+  let counterDisplay = document.querySelector("#counterDisplayID");
+  counterDisplay.innerText = `Displaying ${numberOfMovies} / ${totalMovies} ${tvShowDisplay}`; // update of the counter
+}
+
+let tvShowData = getAllShows().sort(compare);
```

Comment:
You set up a global variable that holds all shows already sorted, but you are still calling `getAllShows().sort(compare)` in 6 other places. It would be better to use that variable or pass it to functions as a parameter.